src/API/api.js

```diff
@@ -1845,6 +1845,37 @@ class API {
 		return this.getPaymentData(...args);
 	}
 
+	/**
+	 * Update currencies to the target
+	 *
+	 * @param {Actor/Token/TokenDocument} target                The actor to update the currencies to
+	 * @param {string} currencies                               A string of currencies to update (eg, "5gp 25sp")
+	 * @param {object} options                                  Options to pass to the function
+	 * @param {string/boolean} [options.interactionId=false]    The ID of this interaction
+	 *
+	 * @returns {Promise<object>}                               An object containing the items and attributes update to the target
+	 */
+	static updateCurrencies(target, currencies, { interactionId = false } = {}) {
+
+		const targetActor = Utilities.getActor(target);
+		const targetUuid = Utilities.getUuid(targetActor);
+		if (!targetUuid) throw Helpers.custom_error(`updateCurrency | Could not determine the UUID, please provide a valid target`);
+
+		if (typeof currencies !== "string") {
+			throw Helpers.custom_error(`updateCurrency | currencies must be of type string`)
+		}
+
+		const currenciesToUpdate = PileUtilities.getPriceFromString(currencies).currencies
+			.filter(currency => currency.quantity);

```

Comment:
If you set the currencies to 0, the thrown error below will cause the method to fail to run, no? We want to be able to set currencies to 0, right?

Reply:
K didn't test with the 0 value... i'll fix it

Reply:
K now this should be good https://github.com/fantasycalendar/FoundryVTT-ItemPiles/pull/505/commits/81e1c7a6d819b8657c602d10721586e31eb07e46 , but i modified the getPriceFromString method with a additional filter on the identifier ? Chek out this linehttps://github.com/fantasycalendar/FoundryVTT-ItemPiles/pull/505/commits/81e1c7a6d819b8657c602d10721586e31eb07e46#diff-a06d7631e0868da987d3a492d985bac32afac7d47915c5586e9dd5f8a1ea0511R879